Python/chapter01/1.1 - Is Unique/miguel_1.1_sol.py

```diff
@@ -0,0 +1,60 @@
+"""
+Python version 3.7.0
+1.1
+Implement an algorithm to determine if a string has all unique characters.
+What if you cannot use additional data structures?
+
+Given:		Expect:
+tacos		true
+swag		true
+bobby		false
+california	false
+orbit		true
+e		    true
+"""
+
+
+def is_unique(input_str):
+    freqs = {}
+    for c in input_str:
+        if c not in freqs:
+            freqs[c] = 1
+        else:
```

Comment:
you can use the [guard clause pattern](http://wiki.c2.com/?GuardClause) here to avoid over-indenting:

```python3
if c in freqs:
  return False
freqs.add(c)
```